src/main/java/core/basesyntax/service/impl/TransactionProcessor.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.TransactionParser;
+import java.util.List;
+
+public class TransactionProcessor {
+    private final OperationStrategy operationStrategy;
+    private final TransactionParser transactionParser;
+
+    public TransactionProcessor(OperationStrategy operationStrategy,
+                                TransactionParser transactionParser) {
+        this.operationStrategy = operationStrategy;
+        this.transactionParser = transactionParser;
+    }
+
+    public void processTransactions(List<String> inputData) {
+        inputData.stream()
+                .map(transactionParser::parseLine)
+                .forEach(transaction -> {
+                    OperationHandler handler = operationStrategy
+                            .getHandler(transaction.getOperation());
+                    Fruit fruit = new Fruit(transaction.getFruitName());
+                    handler.apply(fruit, transaction.getQuantity());
```

Comment:
Creating a new Fruit object within the processing loop violates the encapsulation principle. Instead, the Fruit object should be retrieved or created in a service dedicated to managing fruits, ensuring that the same Fruit instance is used for all operations on that particular fruit.

Reply:
The logic was refactored